Add Package Control docs (#134)
sublimetext-io/docs.sublimetext.io: docs/reference/pc_channels_repositories.md

---
title: "Package Control: Channels and Repositories"
---

<!-- Originals: -->
<!-- https://packagecontrol.io/docs/channels_and_repositories -->
<!-- https://github.com/wbond/packagecontrol.io/blob/master/app/html/docs/channels_and_repositories.html -->

# Channels and Repositories

Channels contain a list of one or more repositories. Repositories contain a list of one or more packages.

Channels → Repositories → Packages.

> The [package_control_channel][pcc] git repo collects lists of packages, and various repositories (like those for SublimeLinter or LSP plugins). This creates the primary public repository for Sublime Text packages. Each git repository is then crawled for additional metadata, and this is all compiled into the public [channel.json][asset]. 

Reviewer: ```suggestion
The [`package_control_channel` Git repository][pcc]  collects lists of packages and various repositories (like those for SublimeLinter or LSP plugin famlies). This creates the primary public repository for Sublime Text packages. Each Git repository is then crawled for additional metadata, and this is all compiled into the public [`channel.json`][asset]. 
```

The Package Control client comes pre-installed with this channel.
(Well, currently it still comes with the old one: https://packagecontrol.io/channel_v3.json).

[pcc]: https://github.com/wbond/package_control_channel
[asset]: https://github.com/packagecontrol/thecrawl/releases/tag/the-channel

## More about channels

Channels are JSON files hosted on a URL that contains a list of repository URLs. When a user requests to install a package, their channels are queried for a list of repositories.  

> See the [channel.json documentation][channel] for the format. 

Reviewer: ```suggestion
See the [`channel.json` documentation][channel] for the format. 
```

[channel]: pc_channel.html

## More about repositories

Repositories are JSON files hosted on a URL that contain a list of packages.

> See the [repository.json documentation][repo] for the format. 

Reviewer: ```suggestion
See the [`repository.json` documentation][repo] for the format. 
```
